numpy/ma/tests/test_core.py

```diff
@@ -1990,6 +2001,205 @@ def test_datafriendly_mul_arrays(self):
         assert_equal(a, [[1, 1], [3, 3]])
         assert_equal(a.mask, [[0, 1], [0, 1]])
 
+    def test_inplace_addition_scalar_type(self):
+        # Test of inplace additions
+        for t in self.othertypes:
```

Comment:
Maybe as a pointer, could you please let me know if there is a good way to verify the casting? Would `numpy.can_cast` work in this case? If so, how strict of a setting would I need to use to reproduce the correct behavior.